Earthfile

```
@@ -104,9 +104,17 @@ earthly-script-no-stdout:
     FROM earthly/dind:alpine
     RUN apk add --no-cache --update bash
     COPY earthly .earthly_version_flag_overrides .
+
+    # TODO this script performs an explicit "docker pull earthlybinaries:prerelease" which can cause rate-limiting
+    # to work-around this, we will copy an earthly binary in, and update the last-check state files
+    COPY +for-linux/earthly /root/.earthly/earthly-prerelease
+    RUN echo "$(date +%s)" >/tmp/last-earthly-prerelease-check
+    RUN echo "$(md5sum .earthly_version_flag_overrides | awk '{print $1}')" > /tmp/last-earthly-flag-hash
+
     WITH DOCKER --pull earthly/earthlybinaries
```

BTW shouldn't this be

```suggestion
    WITH DOCKER --pull earthly/earthlybinaries:prerelease
```

---

since we're skipping the part that pulls via docker, we can remove the entire `WITH docker`.